Explorer/Assets/DCL/Multiplayer/Connections/Messaging/MessageWrap.cs

```diff
@@ -33,27 +37,68 @@ public MessageWrap(T payload, IDataPipe dataPipe, List<string> recipients, IMult
             sent = false;
         }
 
-        public void Send(DataPacketKind dataPacketKind = DataPacketKind.KindLossy)
+        public async UniTaskVoid SendAndDisposeAsync(DataPacketKind dataPacketKind = DataPacketKind.KindLossy)
```

Comment:
I believe we should provide a cancellation token (and gracefully check it after `SwitchToThreadPool`) to prevent exceptions in the edge cases on application exit